gen/src/main/java/com/megaease/easeagent/gen/AssemblyProcessor.java

```diff
@@ -77,6 +87,30 @@ private void process(final ProcessUtils utils, final Assembly assembly, String p
         });
     }
 
+    private TreeSet<String> loadProviders() {
+        ClassLoader loader = this.getClass().getClassLoader();
+        final String providers = "META-INF/services/" + Transformation.class.getCanonicalName();
+        Enumeration<URL> configs;
+        TreeSet<String> pset = new TreeSet<>();
+        try {
+            configs = loader.getResources(providers);
+            while (configs.hasMoreElements()) {
+                URL path = configs.nextElement();
+                InputStream in = path.openStream();
+                BufferedReader r = new BufferedReader(new InputStreamReader(in, StandardCharsets.UTF_8));
+                String line;
+                while((line = r.readLine()) != null) {
+                    pset.add(line);
+                }
+                r.close();
+                in.close();
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
```

Comment:
The exception should be thrown.

Reply:
it is not an error when there is not file found, so maybe we should delete the 'e.printStackTrace()' here.

Reply:
Is there a problem running the agent if the file is not found? Are the files not found the ones that are needed?